src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -884,7 +884,87 @@ public bool Remove(TKey key, out TValue value)
             return false;
         }
 
-        public bool TryGetValue(TKey key, out TValue value)
+		public int RemoveAll(Predicate<KeyValuePair<TKey, TValue>> match)
+		{
+			if (match == null)
+			{
+				ThrowHelper.ThrowArgumentNullException(ExceptionArgument.match);
+			}
+
+			int removedCount = 0;
+			int count = _count;
+			Entry[] entries = _entries;
+			int[] buckets = _buckets;
+			for (int entryIndex = 0; entryIndex < count; entryIndex++)
+			{
+				if (entries[entryIndex].hashCode >= 0)
+				{
+					int version = _version;
+					TKey key = entries[entryIndex].key;
+					if (match(new KeyValuePair<TKey, TValue>(key, entries[entryIndex].value)))
+					{
+						if (version!=_version)
+						{
+							ThrowHelper.ThrowInvalidOperationException_InvalidOperation_EnumFailedVersion();
+						}
+
+						int collisionCount = 0;
+						int hashCode = (_comparer?.GetHashCode(key) ?? key.GetHashCode()) & 0x7FFFFFFF;
```

Comment:
This looks like a exact copy of `Remove(TKey key)`. Call that instead of duplicating the code here. Yes, it will be a tad less efficient in microbenchmarks which is fine.

Reply:
Note if you do call Remove, you will need to update your stored _version here because Remove will incrment it.

Reply:
I've removed the duplication of the Remove body and simply called Remove(). Benchmarking shows a tiny difference (scaled 0.15 => 0.16) but I think that's insignificant in comparison to the code size change.